litgpt/data/llama_data.py

```diff
@@ -0,0 +1,104 @@
+# Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Optional, Union
+
+from torch.utils.data import DataLoader
+
+from litgpt import Tokenizer
+from litgpt.data import DataModule
+
+
+@dataclass
+class LlamaDataModule(DataModule):
```

Comment:
I would prefer if we keep the TinyLlama data module with the implementation that is there. If I understand correctly, you'd need a way to exclude the starcoder data. So I propose to make these changes directly there so we can see the diff. The micro-llama config could then use the same data module (it could also be renamed, but we can't call it Llama data module because it's not the data that is used to train all llama family models).

Reply:
So basically like 

```python
class TinyLlama(...) # like before but with starcoder option


class MicroLlama(TinyLlama)
def __init__(self...):
    super().__init__(..., use_starcoder=True)
```

?

I can help with that @keeeeenw  ... sorry for sending you the wrong direction  ...

Reply:
I didn't know the suggestion came from you, sorry. We can also keep it like this, I don't have a strong opinion. I would just avoid inheritance abstraction if it's not necessary. MicroLlama as subclass of TinyLlama would definitely work yes, I'd be fine with that.

Reply:
It was mainly to avoid code duplication but yeah, maybe "MicroLlama as subclass of TinyLlama" is even better :)

Reply:
No worries @rasbt ! I appreciate the feedback from you and @awaelchli. I updated the code to use ```class MicroLlama(TinyLlama)``` before I saw your comment. It looks like we are aligned now. This will give us the diff for TinyLlama and remove the need for LlamaDataModule.